zilencer/management/commands/populate_db.py

```diff
@@ -232,31 +236,64 @@ def handle(self, **options: Any) -> None:
+                    'Cordelia Lear': ['Verona'],
+                    'King Hamlet': ['Verona', 'Denmark'],
+                    'Iago': ['Verona', 'Denmark', 'Scotland'],
+                    'Othello, the Moor of Venice': ['Verona', 'Denmark', 'Scotland'],
+                    'Prospero from The Tempest': ['Verona', 'Denmark', 'Scotland', 'Venice'],
+                    'Zoe': ['Verona', 'Denmark', 'Scotland', 'Venice', 'Rome']
+                }
+
+                for i, profile_name in enumerate(subscriptions_map.keys()):
+                    profile = UserProfile.objects.get(full_name=profile_name)
+                    for stream_name in subscriptions_map[profile_name]:
+                        stream = Stream.objects.get(name=stream_name)
+                        r = Recipient.objects.get(type=Recipient.STREAM, type_id=stream.id)
+                        s = Subscription(
+                            recipient=r,
+                            user_profile=profile,
+                            color=STREAM_ASSIGNMENT_COLORS[i % len(STREAM_ASSIGNMENT_COLORS)])
+
+                        subscriptions_to_add.append(s)
+
+                        log = RealmAuditLog(realm=profile.realm,
+                                            modified_user=profile,
+                                            modified_stream_id=stream.id,
+                                            event_last_message_id=0,
+                                            event_type='subscription_created',
+                                            event_time=event_time)
+                        all_subscription_logs.append(log)
+            else:
+                profiles = UserProfile.objects.select_related().filter(
+                    is_bot=False, is_guest=False).order_by("email")  # type: Sequence[UserProfile]
+                for i, profile in enumerate(profiles):
+                    # Subscribe to some streams.
+                    for type_id in recipient_streams[:int(len(recipient_streams) *
+                                                          float(i)/len(profiles)) + 1]:
+                        r = Recipient.objects.get(type=Recipient.STREAM, type_id=type_id)
+                        s = Subscription(
+                            recipient=r,
+                            user_profile=profile,
+                            color=STREAM_ASSIGNMENT_COLORS[i % len(STREAM_ASSIGNMENT_COLORS)])
+
+                        subscriptions_to_add.append(s)
+
+                        log = RealmAuditLog(realm=profile.realm,
+                                            modified_user=profile,
+                                            modified_stream_id=type_id,
+                                            event_last_message_id=0,
+                                            event_type='subscription_created',
+                                            event_time=event_time)
+                        all_subscription_logs.append(log)
```

Comment:
It feels like there's a lot of copied code between the two code paths here; might be cleaner to just make this code generate a `subscriptions_map` and then have the two code paths be identical.